gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/writer/FileAwareInputStreamDataWriterTest.java

```diff
@@ -481,4 +499,15 @@ public void cleanup() {
     }
   }
 
+  protected static class TestFileAwareInputStreamDataWriter extends FileAwareInputStreamDataWriter {
+
+    static boolean isAclSet = false;
```

Comment:
this is very weak verification... also the class-level `static` is an anti-pattern in tests (as most framework runners can work concurrently).

better would be an in-memory map from filepath to a set of acls.  the verification code would ask this mock object to provide the acls it recorded for each path, and then verification would ensure those meet the expectations

Reply:
I have added the verification logic based on your advice in my latest commit!